File: Codes/Hash/Non-Cryptographic/BKDRHash/BKDRHash.cs
/*
	Hash function by Brian Kernighan & Dennis Ritchie
    Archive of Reversing.ID
    Non-Cryptographic Hash
*/

// Fungsi ini merupakan method dari sebuah class
public static uint BKDRHash (string key)
{
    uint seed  = 131;
	uint state = 0;
	uint i = 0;

	for (i = 0; i < key.Length; i++)
	{
		state = (state * seed) + ((byte)key[(int)i]);
	}

	return state;
}

File: Codes/Hash/Non-Cryptographic/BKDRHash/BKDRHash.vb
'  Hash function by Brian Kernighan & Dennis Ritchie
'  Archive of Reversing.ID
'  Non-Cryptographic Hash
'

' Fungsi ini merupakan method dari sebuah class
Public Shared Function BKDRHash (key As String) As UInteger
    Dim seed As UInteger = 131
	Dim state As ULong = 0
	Dim i As UInteger = 0

	For i = 0 To key.Length - 1
		state = ((state * seed) + CByte(AscW(key(CInt(i)))) And UInteger.MaxValue)
	Next

	Return state
End Function